powersimdata/input/change_table.py

```diff
@@ -462,31 +462,58 @@ def scale_congested_mesh_branches(self, ref_scenario, **kwargs):
         """
         scale_congested_mesh_branches(self, ref_scenario, **kwargs)
 
-    def add_storage_capacity(self, bus_id):
+    def add_storage_capacity(self, info):
         """Sets storage parameters in change table.
 
-        :param dict bus_id: key(s) for the id of bus(es), value(s) is (are)
-            capacity of the energy storage system in MW.
-        :raises TypeError: if bus_id is not a dict.
-        :raises ValueError: if bus_id contains any bus ids not present in the grid,
-            or any non-positive values are given.
+        :param list info: each entry is a dictionary. The dictionary gathers
+            the information needed to create a new storage device.
+        :raises TypeError: if info is not a list.
+        :raises ValueError: if any of the new storages to be added have bad values.
         """
-        if not isinstance(bus_id, dict):
-            raise TypeError("bus_id must be a dict")
+        if not isinstance(info, list):
+            raise TypeError("Argument enclosing new storage(s) must be a list")
+
+        info = copy.deepcopy(info)
+        new_storages = []
+        required = {"bus_id", "capacity"}
+        optional = {
+            "duration",
+            "min_stor",
+            "max_stor",
+            "energy_value",
+            "InEff",
+            "OutEff",
+            "LossFactor",
+            "terminal_min",
+            "terminal_max",
+        }
         anticipated_bus = self._get_new_bus()
-        diff = set(bus_id.keys()).difference(set(anticipated_bus.index))
-        if len(diff) != 0:
-            raise ValueError(f"No bus with the following id: {', '.join(diff)}")
-        for k, v in bus_id.items():
-            if not isinstance(v, (float, int)):
-                raise ValueError(f"values must be numeric, bad type for bus {k}")
-            if v <= 0:
-                raise ValueError(f"values must be positive, bad value for bus {k}")
+        for i, storage in enumerate(info):
+            self._check_entry_keys(storage, i, "storage", required, None, optional)
+            if storage["bus_id"] not in anticipated_bus.index:
+                raise ValueError(
+                    f"No bus id {storage['bus_id']} available for {ordinal(i)} storage"
+                )
+            for o in optional:
+                if o not in storage:
+                    storage[o] = self.grid.storage[o]
```

Comment:
We should keep in mind that this won't work if we add storage in the base grid (although very unlikely).

Reply:
What do you mean?

Reply:
If we have storage devices in the base grid, it will be listed in grid.storage right? And in that case, grid.storage will be no longer default values, right?

Reply:
Storage devices that are part of the base grid will have entries in `grid.storage["gen"]`, `grid.storage["StorageData"]`, etc., but I believe `grid.storage` would still have top-level keys that give some default parameters for construction of new entries (like we have when we load `grid.mat` files that have storage).

Reply:
I guess I'm trying to understand the following situation: if we have a storage device with all the optional parameters set, then they are listed in `grid.storage["StorageData"]` and we still have default values stored in the top level dictionary of `grid.storage`, right?

Reply:
Exactly. You should be able to play around with a scenario in Create state and see exactly that.